monitoring/defender/autotask/on_module_factory_events.js

```diff
@@ -0,0 +1,1222 @@
+        {
+          name: `Invalid Inputs`,
+          value: invalidInputs,
+        },
+      ],
+      url: `${NETWORKS[chainId].networkExplorerUrl}/tx/${txHash}`,
+    },
+  ];
+
+  if ([invalidTimeout, invalidCooldown, invalidBond].includes(true)) {
+    sendNotification = true;
+  }
+
+  if ([isController, isOwner].includes(false)) {
+    sendNotification = true;
+  }
+  return { sendNotification, params };
+};
+
+/**
+ *
+ * @param {*} realityContract
+ * @param {*} provider
+ * @param {string} txHash
+ * @param {number} chainId
+ * @param {string} ensName
+ * @param {string} discordWebHookUrl
+ */
+const handleContractMethods = async (
+  realityContract,
+  provider,
+  txHash,
+  chainId,
+  ensName,
+  discordWebHookUrl
+) => {
+  const avatar = await realityContract.avatar();
+  const isController = await checkIfIsController(provider, ensName, avatar);
+  const isOwner = await checkIfIsOwner(provider, ensName, avatar, chainId);
+  const questionTimeout = await realityContract.questionTimeout();
+  const questionCooldown = await realityContract.questionCooldown();
+  const bond = await realityContract.minimumBond();
+  const { sendNotification, params } = generateDiscordParams(
+    questionTimeout,
+    questionCooldown,
+    bond,
+    isController,
+    isOwner,
+    ensName,
+    chainId,
+    txHash
+  );
+  if (sendNotification) {
+    await axios.post(discordWebHookUrl, params);
+  }
+};
+
+/**
+ *
+ * @param {string} templateQuestionText //Template question text
+ * @returns {string}
+ */
+const getEnsName = (templateQuestionText) => {
+  if (templateQuestionText.includes(".eth")) {
+    const initialTemplateText = templateQuestionText.substr(
+      0,
+      templateQuestionText.indexOf(".eth")
+    );
+    const words = initialTemplateText.split(" ");
+    const ensName = words[words.length - 1];
+    return `${ensName}.eth`;
+  }
+};
+
+exports.handler = async function (event) {
+  console.log(
+    "\n\n/************************** EVENT *****************************/"
+  );
+  console.log(JSON.stringify(event));
+  console.log(
+    "/**********************************************************/\n\n"
+  );
+  if (event && event.request && event.request.body) {
+    // variables from autotask creation
+    const rpcUrl = "{{rpcUrl}}";
+    const discordWebHookUrl = "{{discordWebHookUrl}}";
+    const transaction = event.request.body.transaction;
+    const logs = transaction.logs;
+    const txHash = transaction.transactionHash;
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    const chainId = event.request.body.sentinel.chainId;
+    const utils = ethers.utils;
+    console.log(
+      "\n\n/*********************** TRANSACTION **************************/"
+    );
+    const decoded = await decode(logs, utils);
+    if (decoded) {
+      const { templateQuestionText } = decoded;
+      decoded.proxyAddresses.forEach(async (proxyAddress) => {
+        const ensName = getEnsName(templateQuestionText);
```

Comment:
This is only possible sometimes (it can be a custom template without any related ENS name). Therefore this must also handle the situation where there is no ENS name in the template.

Perhaps `getEnsName` can throw when there is no ENS name, then we handle it with `try` `catch` or something similar.